### ISSUE TITLE
TypeError when using @import for local and remote stylesheets

### ISSUE THREAD

**Issue by ohskylab** [postcss/postcss-import#21]
If I have one or more `@import` pointing to an external URL (e.g. `@import url("//hello.myfonts.net/count/");`) and nothing else, postcss-import doesn't inline them. :thumbsup:

However, if the stylesheet *also* has an `@import` pointing to a local stylesheet for inlining, I get the following:

```bash
/Projects/postcss-import-test/node_modules/postcss/lib/node.js:284
                if (p != root && p.parent && p.parent == root) {
                                  ^
TypeError: Cannot read property 'parent' of undefined
    at /Projects/postcss-import-test/node_modules/postcss/lib/node.js:284:35
    at /Projects/postcss-import-test/node_modules/postcss/lib/container.js:116:26
    at Root.each (/Projects/postcss-import-test/node_modules/postcss/lib/container.js:94:22)
    at Root.eachInside (/Projects/postcss-import-test/node_modules/postcss/lib/container.js:115:21)
    at Declaration.style (/Projects/postcss-import-test/node_modules/postcss/lib/node.js:282:18)
    at Declaration.style (/Projects/postcss-import-test/node_modules/postcss/lib/node.js:362:35)
    at Declaration.stringify (/Projects/postcss-import-test/node_modules/postcss/lib/declaration.js:19:27)
    at Rule.stringifyContent (/Projects/postcss-import-test/node_modules/postcss/lib/container.js:34:27)
    at Rule.stringifyBlock (/Projects/postcss-import-test/node_modules/postcss/lib/container.js:49:18)
    at Rule.stringify (/Projects/postcss-import-test/node_modules/postcss/lib/rule.js:23:14)
```

Also, if the stylesheet to be inlined only contains an empty declaration block (e.g. `p {}`), I don't get the error but do see:

```css
@import url("http://hello.myfonts.net/count/");

undefinedp{}
```
Looks like something is creeping in with parsing, prepending that `undefined`, but haven't been able to get further than that.

Noticed this when bumping a project from `postcss@3.0.0` and `postcss-import@4.1.1` to `postcss@4.0.3` and `postcss-import@5.0.0`. Works as expected (ie. remote stylesheet `@import` ignored, local inlined) with the older versions.

Test case at [this gist](https://gist.github.com/ohskylab/5ea14e230068bcd8cfe4).

**Comment by MoOx** [comment]
This is weird. This looks like a bug not covered by this fixture https://github.com/postcss/postcss-import/blob/master/test/fixtures/ignore.css

**Comment by VinSpee** [comment]
aha! I have been having this same issue and (strangely enough) am also using myfonts.

**Comment by ohskylab** [comment]
@MoOx Yeah, weird! Looking deeper into the cause may be beyond me, but can have a go at writing a fixture to test mixing local and remote `@import` to help catch this in future, if that would be helpful?

Interesting that @VinSpee has the same issue with the same service but possibly not surprising: MyFonts' external `@import` is first I've seen in a long time. Reason for avoiding it in the first place (performance) is still valid, I think. So bit of an edge case, this :)

**Comment by VinSpee** [comment]
And honestly, I just removed the "count" import line and the fonts still load.

**Comment by iamvdo** [comment]
Same error on iamvdo/pleeease#40, related to postcss/postcss-import#22

**Comment by MoOx** [comment]
Can you guys provide me both input(s) and expected output so I can reproduce this bug in a flash ?
Thanks :)

**Comment by VinSpee** [comment]
```
@import url("//hello.myfonts.net/count/2c2962");
```

output:
```
TypeError: Cannot read property 'parent' of undefined
    at /Projects/postcss-import-test/node_modules/postcss/lib/node.js:284:35
    at /Projects/postcss-import-test/node_modules/postcss/lib/container.js:116:26
    at Root.each (/Projects/postcss-import-test/node_modules/postcss/lib/container.js:94:22)
    at Root.eachInside (/Projects/postcss-import-test/node_modules/postcss/lib/container.js:115:21)
    at Declaration.style (/Projects/postcss-import-test/node_modules/postcss/lib/node.js:282:18)
    at Declaration.style (/Projects/postcss-import-test/node_modules/postcss/lib/node.js:362:35)
    at Declaration.stringify (/Projects/postcss-import-test/node_modules/postcss/lib/declaration.js:19:27)
    at Rule.stringifyContent (/Projects/postcss-import-test/node_modules/postcss/lib/container.js:34:27)
    at Rule.stringifyBlock (/Projects/postcss-import-test/node_modules/postcss/lib/container.js:49:18)
    at Rule.stringify (/Projects/postcss-import-test/node_modules/postcss/lib/rule.js:23:14)
```

**Comment by MoOx** [comment]
A fix landed in master. Can you test it guys @ohskylab @vinspee ?

**Comment by ohskylab** [comment]
Yeah, `master` works for me in my tests and actual project. Thank you!

**Comment by MoOx** [comment]
Released as 5.0.1
